### PR TITLE
Separate CommandBox and ResultsDisplay #91

### LINKED ISSUE #91: Separate CommandBox and ResultsDisplay
ResultsDisplay should not be inside CommandBox UiPart. In future, ResultsDisplay can show results of other UI actions, not just commands.

### PR DESCRIPTION
**Pull request by lejolly**
Fixes #91

### CODE REVIEW

**Comment by damithc** [comment]
Ready for review?

**Comment by lejolly** [comment]
yes.

**Comment by damithc** [comment]
Noted. As before, use labels to indicate PR status.

File: src/main/java/seedu/address/commons/events/ui/UpdateResultDisplayEvent.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package seedu.address.commons.events.ui;
+
+import seedu.address.commons.events.BaseEvent;
+
+/**
+ * Indicates a request to post a message to ResultDisplay
+ */
+public class UpdateResultDisplayEvent extends BaseEvent {
```
Comment (damithc) [inline src/main/java/seedu/address/commons/events/ui/UpdateResultDisplayEvent.java]: An event represents something that happened, not something to do. Event handlers will decide what to do with the event. So a better name for this class is 'NewResultAvailableEvent'

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 2]
```diff
@@ -70,15 +68,15 @@ public void setPlaceholder(AnchorPane pane) {
...
 
         /* We assume the command is correct. If it is incorrect, the command box will be changed accordingly
          * in the event handling code {@link #handleIncorrectCommandAttempted}
          */
         setStyleToIndicateCorrectCommand();
-        mostRecentResult = logic.execute(previousCommandTest);
-        resultDisplay.postMessage(mostRecentResult.feedbackToUser);
+        CommandResult mostRecentResult = logic.execute(previousCommandText);
         logger.info("Result: " + mostRecentResult.feedbackToUser);
+        EventsCenter.getInstance().post(new UpdateResultDisplayEvent(mostRecentResult.feedbackToUser));
```
Comment (damithc) [inline src/main/java/seedu/address/ui/CommandBox.java]: There is a `raise()` method in parent class.

**Comment by lejolly** [comment]
changes made.

File: src/main/java/seedu/address/commons/events/ui/NewResultAvailableEvent.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package seedu.address.commons.events.ui;
+
+import seedu.address.commons.events.BaseEvent;
+
+/**
+ * Indicates a new message for ResultDisplay
```
Comment (damithc) [inline src/main/java/seedu/address/commons/events/ui/NewResultAvailableEvent.java]: Should not mention ResultDisplay. An event doesn't know where it is used.

**Comment by damithc** [comment]
I'll merge after updating diagrams.

File: src/main/java/seedu/address/ui/ResultDisplay.java [hunk 4]
```diff
@@ -41,6 +47,7 @@ public void configure() {
         mainPane.getChildren().add(resultDisplayArea);
         FxViewUtil.applyAnchorBoundaryParameters(mainPane, 0.0, 0.0, 0.0, 0.0);
         placeHolder.getChildren().add(mainPane);
+        registerAsAnEventHandler(this);
```
Comment (damithc) [inline src/main/java/seedu/address/ui/ResultDisplay.java]: @lejolly Can some of the stuff in this method be pun in the fxml file instead?
Also, it mentions STATUS_BAR_STYLE_SHEET. Can that be fixed as well?

**Comment by damithc** [comment]
Updated diagram. Waiting @lejolly 's input on another comment.

**Comment by lejolly** [comment]
ok changes made.

**Pull request merged**